python/src/aac/plugins/first_party/lsp_server/providers/rename_provider.py

```diff
@@ -120,7 +130,27 @@ def _get_enum_value_type_text_edits(
         enum_references = get_enum_references_from_context(definition_to_find, language_context)
         enum_references_to_alter = [*enum_references, definition_to_find]
 
-        for definition in enum_references_to_alter:
+        for definition in _filter_editable_definitions(enum_references_to_alter):
+            reference_lexemes = [lexeme for lexeme in definition.lexemes if remove_list_type_indicator(lexeme.value) == old_value]
+            for lexeme_to_replace in reference_lexemes:
+                document_edits = edits.get(lexeme_to_replace.source, [])
+                replacement_range = get_location_from_lexeme(lexeme_to_replace).range
+                document_edits.append(TextEdit(range=replacement_range, new_text=new_value))
+                edits[str(lexeme_to_replace.source)] = document_edits
```

Comment:
We should probably turn this into a separate function and re-use it in both `_get_*_text_edits` functions.

Reply:
Done